packages/create-amplify/src/package-manager-controller/package_manager_controller_factory.ts

```diff
@@ -18,6 +14,27 @@ import {
 } from '../project-initializer/index.js';
 
 export type DependencyType = 'dev' | 'prod';
+type PackageManagerName = 'npm' | 'yarn-classic' | 'yarn-modern' | 'pnpm';
+type PackageManagerExecutable = 'npm' | 'yarn' | 'pnpm';
+type PackageManagerBinaryRunner = 'npx' | 'yarn' | 'pnpm';
+type PackageManagerInstallCommand = 'install' | 'add';
+type PackageManagerLockFile =
+  | 'package-lock.json'
+  | 'yarn.lock'
+  | 'pnpm-lock.yaml';
+type PackageManagerInitDefault = Readonly<string[]>;
+export type PackageManagerProps = {
+  name: PackageManagerName;
+  executable: PackageManagerExecutable;
+  binaryRunner: PackageManagerBinaryRunner;
+  installCommand: PackageManagerInstallCommand;
+  lockFile: PackageManagerLockFile;
+  initDefault: PackageManagerInitDefault;
+};
+
+type PackageManagers = {
+  [key in PackageManagerName]: PackageManagerProps;
+};
```

Comment:
This should be completely removed, not just moved to a different file

Reply:
Could you elaborate where we should put those types? We'd get type error without them

Reply:
We should just remove them. Take `PackageMangerName` for example. That can just be a private instance member of each PackageManagerController instance. It doesn't need a special type as it's just a const string within that class. The same pattern can be used for all the other values as well.

Reply:
After talking offline, I understand that we don't have to use `PackageManagerProps` to keep all the Package Managers using the same pattern, so we can remove those types.